source/extensions/filters/network/thrift_proxy/conn_manager.h

```diff
@@ -87,8 +89,8 @@ class ConnectionManager : public Network::ReadFilter,
 
     // ProtocolConverter
     FilterStatus messageBegin(MessageMetadataSharedPtr metadata) override;
-    FilterStatus fieldBegin(absl::string_view name, FieldType field_type,
-                            int16_t field_id) override;
+    FilterStatus fieldBegin(absl::string_view name, FieldType& field_type,
+                            int16_t& field_id) override;
```

Comment:
What's the scenario where an implementation of `fieldBegin` will need to modify the `field_type` or `field_id`?

Reply:
The use case is some complex filter that's manipulating thrift structs. I could make this simpler by reverting this and not allowing thrift filters to manipulate the request beyond the message metadata.

Reply:
Got it, thanks. I guess making the Thrift structs mutable by filters is consistent with how HTTP filters can mutate message bodies, so changing these parameters to references seems reasonable.